adaptive_navigation/lib/scaffold.dart

```diff
@@ -52,105 +46,132 @@ class AdaptiveNavigationScaffold extends StatefulWidget {
-      );
-    }
-
-    // Show a navigation rail
-    if (_isMediumScreen(context)) {
-      return Scaffold(
-        appBar: AppBar(
-          title: widget.title,
-        ),
-        body: Row(
+        );
+      case NavigationType.drawer:
+        // Show a Row containing a Drawer and Scaffold.
+        return Row(
           children: [
-            NavigationRail(
-              leading: widget.floatingActionButton,
-              destinations: [
-                ...widget.destinations.map(
-                  (d) => NavigationRailDestination(
-                    icon: Icon(d.icon),
-                    label: Text(d.title),
+            Drawer(
+              child: Column(
+                children: [
+                  DrawerHeader(
+                    child: Center(
+                      child: widget.title,
+                    ),
                   ),
-                ),
-              ],
-              selectedIndex: widget.currentIndex,
-              onDestinationSelected: widget.onNavigationIndexChange ?? (_) {},
+                  for (final destination in widget.destinations)
+                    ListTile(
+                      leading: Icon(destination.icon),
+                      title: Text(destination.title),
+                      selected: widget.destinations.indexOf(destination) ==
+                          widget.currentIndex,
+                      onTap: () => _destinationTapped(destination),
+                    ),
+                ],
+              ),
             ),
             VerticalDivider(
               width: 1,
               thickness: 1,
               color: Colors.grey[300],
             ),
             Expanded(
-              child: widget.body,
+              child: Scaffold(
+                appBar: AppBar(),
+                body: widget.body,
+                floatingActionButton: widget.floatingActionButton,
+              ),
             ),
           ],
-        ),
-      );
+        );
     }
-
-    // Show a bottom app bar
-    return Scaffold(
-      body: widget.body,
-      appBar: AppBar(title: widget.title),
-      bottomNavigationBar: BottomNavigationBar(
-        items: [
-          ...widget.destinations.map(
-            (d) => BottomNavigationBarItem(
-              icon: Icon(d.icon),
-              title: Text(d.title),
-            ),
-          ),
-        ],
-        currentIndex: widget.currentIndex,
-        onTap: widget.onNavigationIndexChange,
-      ),
-      floatingActionButton: widget.floatingActionButton,
-    );
   }
 
   void _destinationTapped(AdaptiveScaffoldDestination destination) {
-    var idx = widget.destinations.indexOf(destination);
-    if (idx != widget.currentIndex) {
-      widget.onNavigationIndexChange(idx);
+    final index = widget.destinations.indexOf(destination);
+    if (index != widget.currentIndex) {
+      widget.onNavigationIndexChange(index);
     }
   }
 }
+
+typedef NavigationType NavigationTypeResolver(BuildContext context);
```

Comment:
I like the flexibility here - users can decide when they think the layout should change. I'm wondering if providing the BuildContext is too general. Maybe providing the size of the widget is enough?

Reply:
Thats a good idea, other factors that I can think of that might be used are the Platform type, but that does not require context to compute. One that does might be textScaleFactor though. It might also be better to leave this open and show developers how to get the size with the context in the comments.